frontend/pages/shared-components/text-input/text-input.tsx

```diff
@@ -0,0 +1,49 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons'
+import React, { FormEvent, useState } from 'react'
+import styles from './text-input.module.css'
+import SyntaxHelper from '../../../helpers/syntax-helper'
```

Comment:
Not too much of a fan on this since for me does not separate concerns between functionality of javascript and text UI manipulation which imo should only be through CSS

Reply:
Resolving here to keep the conversation in the related thread above!